### PR TITLE
ConcurrentModificationException during build #1632

### PR DESCRIPTION
**Pull request by wodencafe**
This fixes a potential issue with MarkerSupport in which a ConcurrentModificationException may occur, as per issue #1632. 

This fix affects MarkerSupport.createMarkers, creates a copy of the input parameter Collection<String> msgs, and iterates on the copy rather than the original to avoid the possibility of a ConcurrentModificationException.

### CODE REVIEW

File: bndtools.builder/src/org/bndtools/builder/MarkerSupport.java [hunk 1]
```diff
@@ -79,7 +79,8 @@ void deleteMarkers(String markerType) throws CoreException {
     }
 
     private void createMarkers(Processor model, int severity, Collection<String> msgs, String markerType) throws Exception {
-        for (String msg : msgs) {
+        Collection<String> msgsCopy = new ArrayList<String>(msgs);
+        for (String msg : msgsCopy) {
```
Comment (bjhargrave) [inline bndtools.builder/src/org/bndtools/builder/MarkerSupport.java]: Why not just:

```java
for (String msg : msgs.toArray(new String[0]))
```
? No need to make a new ArrayList. Just the array.

**Comment by wodencafe** [comment]
Hi @bjhargrave , I have made the requested change and updated my commit message :)

**Comment by bjhargrave** [comment]
@wodencafe, Thanks for updating this, but I think your git config is still messed up. See <https://github.com/wodencafe/bndtools/commit/e390eb8d2ddc3975d58c72a9f96a6c38506b9f9c.patch> which shows:

```
From e390eb8d2ddc3975d58c72a9f96a6c38506b9f9c Mon Sep 17 00:00:00 2001
From: cboyd <cboyd@cboyd-kubuntu>
Date: Sun, 5 Feb 2017 00:03:17 -0600
Subject: [PATCH] ConcurrentModificationException during build #1632

Signed-off-by: Chris Boyd <wodencafe@gmail.com>
```

The Signed-off-by looks great but your committer email address is not real. Please properly [configure your user.name and user.email address in git](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup) and amend the commit. Thanks.

**Comment by wodencafe** [comment]
Hi @bjhargrave , I have updated my git config and amended the commit.

**Pull request merged**